controllers/jobs.js

```diff
@@ -0,0 +1,91 @@
+const Jobs = require('../models/Job');
+const handleErrors = require("../utils/parseValidationErrs");
+
+// GET all Jobs for the current user
+const getJobs = async (req, res, next) => {
+    try {
+        const jobs = await Jobs.find({ createdBy: req.user._id });
+        res.render('jobs', { jobs });
+    } catch (error) {
+        handleErrors(error, req, res);
+    }
+};
+
+// POST a new Jobs
+const addJobs = async (req, res, next) => {
+    try {
+        const newJobs = await Jobs.create({ ...req.body, createdBy: req.user._id });
+        res.redirect('/jobs'); 
+    } catch (error) {
+        handleErrors(error, req, res);
+    }
+};
+
+// GET the form for adding a new jobs
+const getNewJobs = async (req, res) => {
+    console.log('getNewJobs called');
+    try {
+        res.render('job', { job: null });
+    } catch (error) {
+        handleErrors(error, req, res);
+    }
+};
+
+// GET a specific jobs for editing
+const editJobs = async (req, res, next) => {
+    try {
+        const jobs = await Jobs.findOne({ _id: req.params.id, createdBy: req.user._id });
```

Comment:
Nitpick:
Just like previous assignments, it’s best to check that `req.params.id` is a valid mongoose id before using it in a “findOne” method.